modules/product/src/Form/AddToCartForm.php

```diff
@@ -105,16 +111,24 @@ public function buildForm(array $form, FormStateInterface $form_state, $product
       '#suffix' => '</div>',
     ];
     $form = $this->variationElement($form, $form_state);
-    $form['quantity'] = [
-      '#type' => 'number',
-      '#title' => $this->t('Quantity'),
-      '#default_value' => $settings['default_quantity'],
-      '#min' => 1,
-      '#max' => 9999,
-      '#step' => 1,
-      '#access' => !empty($settings['show_quantity']),
-      '#weight' => 99,
-    ];
+
+    /** @var \Drupal\commerce_product\Entity\ProductType $product_type */
+    $product_type = $this->entityTypeManager
+      ->getStorage('commerce_product_type')
+      ->load($product->bundle());
+    /** @var \Drupal\commerce_product\Entity\ProductVariationType $variation_type */
+    $variation_type = $this->entityTypeManager
+      ->getStorage('product_variation_type')
+      ->load($product_type->getVariationType());
+
+    $line_item_stub = LineItem::create([
+      'type' => $variation_type->getLineItemType(),
+    ]);
+
+    $form_display = EntityFormDisplay::collectRenderDisplay($line_item_stub, self::LINE_ITEM_FORM_DISPLAY);
```

Comment:
According to ContentEntityForm, once we have $form_display we should stick it in $form_state so that we don't need to repeat this in validate/submit